automated install/basic-install.sh

```diff
@@ -1825,7 +1832,51 @@ configureFirewall() {
     printf "  %b Skipping firewall configuration\\n" "${INFO}"
 }
 
-#
+replaceAddCreateRename() {
+    # origVariable="$1"
+    # newVariable="$2"
+    # theFile="$3"
+    # If the file does not exist and is not larger than 0 bytes, create it with 644 permisions or add 1 byte. sed cannot work on empty files
+    if [[ ! -s "$3" ]] ; then
+       (umask 022 ; echo >> "$3")
+    fi
+    # Replace the value of a variable or add it to a target file if it is not present, preserving indentation and comments.
+    sed -i '/^[#]*[ \t]*'"$2"'=/ {h;s/=[^#]*\(.*\)/='"${!1}"'\1/;s/'"${!1}"'#/'"${!1}"' #/};${x;/^$/{s//'"$2"'='"${!1}"'/;H};x}' $3
+
+}
+
+replaceAddCreate() {
+    # theVariable="$1"
+    # theValue="${!1}"
+    # theFile="$2"
+    # If the file does not exist and is not larger than 0 bytes, create it with 644 permisions or add 1 byte. sed cannot work on empty files
+    if [[ ! -s "$2" ]] ; then
+       (umask 022 ; echo >> "$2")
+    fi
+    # Replace the value of a variable or add it to a target file if it is not present, preserving indentation and comments.
+    sed -i '/^[#]*[ \t]*'"$1"'=/ {h;s/=[^#]*\(.*\)/='"${!1}"'\1/;s/'"${!1}"'#/'"${!1}"' #/};${x;/^$/{s//'"$1"'='"${!1}"'/;H};x}' $2
```

Comment:
Break this in to multiple steps. No need to oneline it.

Reply:
done